docs/benchmarks.md

```diff
@@ -0,0 +1,90 @@
+
+## Benchmarking Goals
+
+1. Establish a baseline performance for `runwasi` shims
+1. Measure the impact of changes to the source code and version upgrades on performance
+1. Test the density of `runwasi` shims in a single node.
+1. Visualize the performance data over time.
+1. Identify areas for further optimizations
+
+## Approaches
+
+### 1. Baseline Performance
+
+To establish a baseline performance, we use the "ghcr.io/containerd/runwasi/wasi-demo-app:latest" and "ghcr.io/containerd/runwasi/wasi-demo-oci:latest" images. These images are simple demo apps that echoes `"hello"` to the console using WASI. We run end-to-end benchmarks for `wasmtime`, `wasmedge`, `wasmer` and `wamr` runtimes and measure the execution duration. The result can be found in [dev/bench](https://containerd.github.io/runwasi/dev/bench) website, under the "Criterion.rs Benchmark" section.
+
+The source code for the benchmarks can be found in [benchs/containerd-shim-benchmarks](https://github.com/containerd/runwasi/tree/containerd-shim-wasm/v0.9.0/benches/containerd-shim-benchmarks) directory.
+
+### 2. Stress Test
+
+[stress-test](https://github.com/containerd/runwasi/tree/containerd-shim-wasm/v0.9.0/crates/stress-test) is a CLI that runs a stress test for the `runwasi` shims. It allows us to deploy a large number of Wasm-based tasks to evaluate task throughput (number of tasks finished per second) under high-density environments. The stress test supports running workloads with mocked containerd server and a real containerd server and supports running `N` tasks in parallel with `M` concurrency. The result can be found in [dev/bench](https://containerd.github.io/runwasi/dev/bench) website, under the "Stress Test Benchmark" section.
+
+### 3. Memory
+
+We've added functions to collect memory usage data from the shim and zygote processes. We use `scripts/benchmark-mem.sh` script to collect the data. The result can be found in[dev/bench](https://containerd.github.io/runwasi/dev/bench) website, under the "Criterion.rs Benchmark" section.
+
+### 4. Networking
+
+This is for Wasmtime shim. We use `hey` to load test the Wasmtime shim running "ghcr.io/containerd/runwasi/wasi-demo-http:latest" workload. We measure the HTTP throughput and latency of the workload. The result can be found in [dev/bench](https://containerd.github.io/runwasi/dev/bench) website, under the "HTTP Throughput" and "HTTP Latency" sections.
+
+### 5. Runwasi vs. Native Execution
+
+We built a distroless container image with `wasmtime` runtime installed. Then we run stress-test for `wasmtime` shim and the distroless container for the same workload. The result is shown below.
```

Comment:
I think this section can be updated now.

Reply:
But I'm happy for this to be merged as is, we can update it once we add the native stress testing to CI.